File: guias/guia2/Point.java
package guia2;

public class Point {
    private double x;
    private double y;
    private double z;
    private double distToOrigin;

    public Point(double x, double y, double z) {
        this.x = x;
        this.y = y;
        this.z = z;

        this.computeDistToOrigin();
    }

    private void computeDistToOrigin() {
        this.distToOrigin = Math.sqrt(Math.pow(this.x, 2) + Math.pow(this.y, 2) + Math.pow(this.z, 2));
    }

    public double getDistToOrigin() {
        return this.distToOrigin;
    }

    @Override
    public String toString() {
        return ("(" + this.x + ", " + this.y + ", " + this.z + ")");
    }
}

File: guias/Utils/Point.kt
package Utils

import kotlin.math.pow
import kotlin.math.sqrt

class Point(private val x: Double, private val y: Double, private val z: Double) {
    var distToOrigin = 0.0
        private set

    private fun computeDistToOrigin() {
        distToOrigin = sqrt(
            x.pow(2.0) + y.pow(2.0) + z.pow(2.0)
        )
    }

    override fun toString(): String {
        return "($x, $y, $z)"
    }

    init {
        computeDistToOrigin()
    }

}